### PR TITLE
Cvx3crv oracle data feed validation fix

### PR DESCRIPTION
**Pull request by iamsahu**
## Goal
Fixed the issue #345 reported in the code4rena competition report [136](https://github.com/code-423n4/2022-01-yield-findings/issues/136)
## Fix
Implemented data feed validation similar to what is implemented in the chainlink oracle

### CODE REVIEW

**Review by alcueca — CHANGES_REQUESTED** [review]
Minor gas thing.

File: contracts/oracles/convex/Cvx3CrvOracle.sol [hunk 1]
```diff
@@ -105,11 +105,29 @@ contract Cvx3CrvOracle is IOracle, AccessControl {
...
+        uint80 roundId;
+        uint80 answeredInRound;
+        int256 daiPrice;
+        int256 usdcPrice;
+        int256 usdtPrice;
+
+        // DAI Price
+        (roundId, daiPrice, , updateTime, answeredInRound) = DAI.latestRoundData();
+        require(daiPrice > 0, "Chainlink DAI price <= 0");
+        require(updateTime != 0, "Incomplete round for DAI");
```
[the remark this answers is not in the document]
Comment (iamsahu) [reply contracts/oracles/convex/Cvx3CrvOracle.sol:120]: Auditors says the opposite.

- https://github.com/code-423n4/2022-01-yield-findings/issues/57

File: contracts/oracles/convex/Cvx3CrvOracle.sol [hunk 2]
```diff
@@ -120,11 +120,29 @@ contract Cvx3CrvOracle is IOracle, AccessControl {
...
+        // DAI Price
+        (roundId, daiPrice, , updateTime, answeredInRound) = DAI.latestRoundData();
+        require(daiPrice > 0, "Chainlink DAI price <= 0");
+        require(updateTime > 0, "Incomplete round for DAI");
+        require(answeredInRound >= roundId, "Stale price for DAI");
+
+        // USDC Price
+        (roundId, usdcPrice, , updateTime, answeredInRound) = USDC.latestRoundData();
+        require(usdcPrice > 0, "Chainlink USDC price <= 0");
+        require(updateTime != 0, "Incomplete round for USDC");
```
Comment (alcueca) [inline contracts/oracles/convex/Cvx3CrvOracle.sol:139]:
```suggestion
        require(updateTime > 0, "Incomplete round for USDC");
```

File: contracts/oracles/convex/Cvx3CrvOracle.sol [hunk 3]
```diff
@@ -120,11 +120,29 @@ contract Cvx3CrvOracle is IOracle, AccessControl {
...
+
```
Comment (alcueca) [inline contracts/oracles/convex/Cvx3CrvOracle.sol:140]:
```suggestion
        require(answeredInRound >= roundId, "Stale price for USDC");

```

File: contracts/oracles/convex/Cvx3CrvOracle.sol [hunk 4]
```diff
@@ -120,11 +120,29 @@ contract Cvx3CrvOracle is IOracle, AccessControl {
...
+        // USDT Price
+        (roundId, usdtPrice, , updateTime, answeredInRound) = USDT.latestRoundData();
+        require(usdtPrice > 0, "Chainlink USDT price <= 0");
+        require(updateTime != 0, "Incomplete round for USDT");
```
Comment (alcueca) [inline contracts/oracles/convex/Cvx3CrvOracle.sol:144]:
```suggestion
        require(updateTime > 0, "Incomplete round for USDT");
```

**Review by alcueca — CHANGES_REQUESTED** [review]
Please fix for consistency.

**Pull request merged**